src/main/java/hotel/models/Hotel.java

```diff
@@ -0,0 +1,69 @@
+package hotel.models;
+
+import customexceptions.InvalidHotelActionException;
+
+import java.time.LocalDate;
+import java.util.*;
+
+public class Hotel {
+	private static final List<Integer> TAKEN_ROOM_NUMBERS = new ArrayList<>();
+	private String name;
+	private List<Room> rooms;
+
+	/**
+	 * Only constructor of the Hotel class
+	 *
+	 * @param name  <b>name</b> of the hotel
+	 * @param rooms an array list consisting of all of the <b>rooms</b>z
+	 */
+	public Hotel(String name, ArrayList<Room> rooms) {
+		this.name = name;
+		this.rooms = rooms;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public List<Room> getRooms() {
+		return rooms;
+	}
+
+	public static List<Integer> getTakenRoomNumbers() {
+		return TAKEN_ROOM_NUMBERS;
+	}
+
+	/**
+	 * Gets a map of key value pairs -
+	 * the room and the free dates for it
+	 *
+	 * @param fromDate     check from date
+	 * @param toDate       check to date
+	 * @param numberOfDays for number of days
+	 * @param numberOfBeds for number of people
+	 * @return returns map consisting only of rooms
+	 * which are found to be vacant in at least one
+	 * date of the specified interval
+	 * @throws InvalidHotelActionException if the number of days does not
+	 *                                     correspond with the specified interval
+	 */
+	public Map<Room, List<LocalDate>> getFreeRoomsAndDates(LocalDate fromDate, LocalDate toDate,
```

Comment:
Sometimes there big structure like Map<SomeClass, List<Map<KeyClass, Map<otherKey, ValueClass>>>. To be more readable there is recommendation to extract them in `wrapper classes`